### PR TITLE
PMM-7 Remove go tip runs, use go version for CI from go.mod file

### PR DESCRIPTION
**Pull request by artemgavrilov**
PMM-7

This PR:
1. Fixes problem with caches overlapping
2. Removes go tip weekly builds
3. Adds missing go build caches
4. Adds verbosity to modules downloading
5. Fetches go runtime version from go.mod file

### CODE REVIEW

File: .github/workflows/admin.yml [hunk 1]
```diff
@@ -47,31 +34,29 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Set up Go release
-        if: env.GO_VERSION != 'tip'
         uses: actions/setup-go@v4
         with:
-          go-version: ${{ env.GO_VERSION }}
+          go-version-file: ${{ github.workspace }}/go.mod
```
Comment (artemgavrilov) [inline .github/workflows/admin.yml:39]: Fetches go version directly from go.mod file

File: .github/workflows/admin.yml [hunk 2]
```diff
@@ -47,31 +34,29 @@ jobs:
...
           path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ env.GO_VERSION }}-modules-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-${{ env.GO_VERSION }}-modules-
+          key: ${{ runner.os }}-go-modules-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-go-modules-
 
       - name: Download Go modules
-        run: go mod download
+        run: go mod download -x
```
Comment (artemgavrilov) [inline .github/workflows/admin.yml:59]: Enables verbose mode (i.e. it will print what it downloads)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -10,58 +10,44 @@ on:
...
 
-      - name: Set up Go tip
-        if: env.GO_VERSION == 'tip'
-        run: |
-          git clone --depth=1 https://go.googlesource.com/go $HOME/gotip
-          cd $HOME/gotip/src
-          ./make.bash
-          echo "GOROOT=$HOME/gotip" >> $GITHUB_ENV
-          echo "$HOME/gotip/bin" >> $GITHUB_PATH
+      - name: Enable Go build cache
```
Comment (artemgavrilov) [inline .github/workflows/main.yml:30]: Build cache saves almost two minutes in this workflow

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -98,7 +84,7 @@ jobs:
         uses: reviewdog/action-golangci-lint@v2
         with:
           github_token: ${{ secrets.ROBOT_TOKEN || secrets.GITHUB_TOKEN }}
-          go_version: ${{ env.GO_VERSION }}
+          go_version_file: ${{ github.workspace }}/go.mod
```
Comment (artemgavrilov) [inline .github/workflows/main.yml:87]: Similar to `setup-go`. Reviewdog will fetch go version from go.mod file.

File: .github/workflows/admin.yml [hunk 5]
```diff
@@ -47,31 +34,29 @@ jobs:
...
+            ${{ runner.os }}-go-build-
 
       - name: Enable Go modules cache
         uses: actions/cache@v3
         with:
           path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ env.GO_VERSION }}-modules-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-${{ env.GO_VERSION }}-modules-
+          key: ${{ runner.os }}-go-modules-${{ hashFiles('**/go.sum') }}
```
Comment (artemgavrilov) [inline .github/workflows/admin.yml:55]: This is why we can't use embedded in `setup-go` action cache. Our cache uses all `go.mod` files in the project, while setup-go can use only one file. However this feature may be implemented soon: https://github.com/actions/setup-go/issues/371

File: .github/workflows/admin.yml (hunk 1 above)
Comment (BupycHuk) [reply .github/workflows/admin.yml:39]: Super

**Review by BupycHuk — APPROVED** [review]

**Review by atymchuk — APPROVED** [review]

**Pull request merged**